components/encryption/src/manager/mod.rs

```diff
@@ -809,10 +809,13 @@ impl DataKeyManager {
         if !scan.exists() {
             return Ok(());
         }
-        let mut iter = walkdir::WalkDir::new(scan).into_iter().peekable();
+        let mut iter = walkdir::WalkDir::new(scan)
+            .into_iter()
+            .filter(|e| e.as_ref().map_or(true, |e| !e.path().is_dir()))
+            .peekable();
         while let Some(e) = iter.next() {
             let e = e?;
-            if e.path_is_symlink() {
+            if e.path().is_symlink() {
```

Comment:
question: what's the difference between e.path_is_symlink() and e.path().is_symlink()

Reply:
Implementation is different. Using std implementation is more consistent.